pkg/registry/core/rest/storage_core.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package master
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/apiserver"
+	"k8s.io/kubernetes/pkg/genericapiserver"
+	kubeletclient "k8s.io/kubernetes/pkg/kubelet/client"
+
+	"k8s.io/kubernetes/pkg/registry/core/componentstatus"
+	configmapetcd "k8s.io/kubernetes/pkg/registry/core/configmap/etcd"
+	controlleretcd "k8s.io/kubernetes/pkg/registry/core/controller/etcd"
+	"k8s.io/kubernetes/pkg/registry/core/endpoint"
+	endpointsetcd "k8s.io/kubernetes/pkg/registry/core/endpoint/etcd"
+	eventetcd "k8s.io/kubernetes/pkg/registry/core/event/etcd"
+	limitrangeetcd "k8s.io/kubernetes/pkg/registry/core/limitrange/etcd"
+	"k8s.io/kubernetes/pkg/registry/core/namespace"
+	namespaceetcd "k8s.io/kubernetes/pkg/registry/core/namespace/etcd"
+	"k8s.io/kubernetes/pkg/registry/core/node"
+	nodeetcd "k8s.io/kubernetes/pkg/registry/core/node/etcd"
+	pvetcd "k8s.io/kubernetes/pkg/registry/core/persistentvolume/etcd"
+	pvcetcd "k8s.io/kubernetes/pkg/registry/core/persistentvolumeclaim/etcd"
+	podetcd "k8s.io/kubernetes/pkg/registry/core/pod/etcd"
+	podtemplateetcd "k8s.io/kubernetes/pkg/registry/core/podtemplate/etcd"
+	"k8s.io/kubernetes/pkg/registry/core/rangeallocation"
+	resourcequotaetcd "k8s.io/kubernetes/pkg/registry/core/resourcequota/etcd"
+	secretetcd "k8s.io/kubernetes/pkg/registry/core/secret/etcd"
+	"k8s.io/kubernetes/pkg/registry/core/service"
+	"k8s.io/kubernetes/pkg/registry/core/service/allocator"
+	etcdallocator "k8s.io/kubernetes/pkg/registry/core/service/allocator/etcd"
+	serviceetcd "k8s.io/kubernetes/pkg/registry/core/service/etcd"
+	ipallocator "k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
+	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
+	serviceaccountetcd "k8s.io/kubernetes/pkg/registry/core/serviceaccount/etcd"
+	utilnet "k8s.io/kubernetes/pkg/util/net"
+)
+
+// LegacyRESTStorageProvider does NOT implement the "normal" RESTStorageProvider (yet!)
```

Comment:
funny comment which follows the "Type does.." style, but only says what it does not ;-) Same below.